src/components/modules/ui.ts

```diff
@@ -132,45 +132,58 @@ export default class UI extends Module {
    * Making main interface
    */
   public async prepare(): Promise<void> {
-    /**
-     * Detect mobile version
-     */
-    this.checkIsMobile();
-
-    /**
-     * Make main UI elements
-     */
-    await this.make();
-
-    /**
-     * Loader for rendering process
-     */
-    this.addLoader();
-
-    /**
-     * Append SVG sprite
-     */
-    await this.appendSVGSprite();
-
-    /**
-     * Make toolbar
-     */
-    await this.Editor.Toolbar.make();
 
-    /**
-     * Make the Inline toolbar
-     */
-    await this.Editor.InlineToolbar.make();
-
-    /**
-     * Load and append CSS
-     */
-    await this.loadStyles();
+    // prepare with read-only state from config
+    this.toggleReadOnly(this.config.readOnly);
+  }
 
-    /**
-     * Bind events for the UI elements
-     */
-    await this.bindEvents();
+  /**
+   * Set or toggle read-only state
+   */
+  public toggleReadOnly(readOnlyEnabled: boolean) {
+    if (readOnlyEnabled) {
+      // Do nothing
+    } else {
```

Comment:
not all the function calls inside the UI prepare method must be in toggleReadOnly, but those modules that won't be used

Reply:
After some experimenting, it looks like `bindEvents` is the only one that must be toggle on/off. We'll need to add an `unbindEvents` method.